tests/detectors/test_detectors_base.py

```diff
@@ -133,6 +133,13 @@ def test_matchtype_with_case_sensitivity(matchtype):
                 False,
             ),  # shouldn't match case sensitive
         ],
+        "startswith": [
+            (f"{lowercase_test_word} start", True),  # should match
+            (
+                f"{lowercase_test_word.upper()} start",
+                False,
+            ),  # shouldn't match case sensitive
+        ],
```

Comment:
This addition would is not yet actually run in the test suite, line 113 above needs to be expanded to include `"startswith"` as a `matchtype`.

Not required for this PR but it would be nice to refactor the tests in this file to consolidate test cases into `@pytest.mark.parametrize()`.

Reply:
Hah, good catch. Indeed - this dict might even be best brought up and made a const, and then `parametrize()` can use its keys. Will take a look.